services/QuillLMS/client/app/bundles/Grammar/actions/session.ts

```diff
@@ -209,7 +215,7 @@ export const getQuestionsForConcepts = (concepts: any, flag: string) => {
   return (dispatch, getState) => {
     dispatch(setSessionPending(true))
     const conceptUIDs = Object.keys(concepts)
-    QuestionApi.getAll(GRAMMAR_QUESTION_TYPE).then((questions) => {
+    QuestionApi.getAll(getUID()).then((questions) => {
```

Comment:
This function is used when the session is either a) a follow-up to a Proofreader activity, or b) when the activity itself has assigned concepts rather than assigned questions ([example activity here](https://www.quill.org/grammar/#/admin/lessons/eUTXuBfu8EpXOq2pSfBpFg/)). In both instances, the questions themselves aren't associated with the activity directly, but instead there are set concepts where questions that are associated with that concept are pulled at random (which is what's happening below, in this method). So I don't think it'll work to pull questions by activity here, because you'll get nothing. We probably need to either move that logic to the backend or leave it alone for now.

Reply:
good catch!